src/main/java/io/zentity/resolution/Query.java

```diff
@@ -133,14 +133,16 @@ public static String makeScriptFieldsClause(Input input, String indexName) throw
      */
     public static String populateMatcherClause(Matcher matcher, String indexFieldName, String value, Map<String, String> params) throws ValidationException {
         String matcherClause = matcher.clause();
+        // Quotes and backslashes need to be escaped for the matcher to correctly perform the substitution
+        String escapedValue = java.util.regex.Matcher.quoteReplacement(Json.jsonStringEscape(value));
         for (String variable : matcher.variables().keySet()) {
             Pattern pattern = matcher.variables().get(variable);
             switch (variable) {
                 case "field":
                     matcherClause = pattern.matcher(matcherClause).replaceAll(indexFieldName);
```

Comment:
https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/regex/Matcher.html using `quoteReplacement` because `replaceAll` has special handling of quotes and dollar signs - this is actually like a partial regex being passed in.

Reply:
LGTM